readabilitySAX.js

```diff
@@ -491,18 +494,18 @@ Readability.prototype.onclosetag = function(tagName){
 	else if(this._settings.cleanConditionally && tagName in cleanConditionally){
 		var p = elem.info.tagCount.p || 0,
 		    contentLength = elem.info.textLength + elem.info.linkLength;
-
+        
 		if(contentLength === 0){
 			if(elem.children.length === 0) return;
 			if(elem.children.length === 1 && typeof elem.children[0] === "string") return;
 		}
 		if((elem.info.tagCount.li - 100) > p && tagName !== "ul" && tagName !== "ol") return;
-		if(contentLength < 25 && (!("img" in elem.info.tagCount) || elem.info.tagCount.img > 2) ) return;
+		if(contentLength < 25 && ((!("img" in elem.info.tagCount) && !("iframe" in elem.info.tagCount)) || elem.info.tagCount.img > 2 || elem.info.tagCount.iframe > 2) ) return;
```

Comment:
Reg: image selection part of the algorithm -- I know it was very hard to understand the logic. It could be simplified, but most heuristics implementation would look similar ugly.

Only questionable aspect was why 25 content length. Probably we can modify algorithm to 2 pass --

1) In pass 1 - collect average statistics for the page. For example, average length of text for image on that particular page.
2) In pass 2 - use these statistics to score images.

Probably such statistics based calculation may cover wider cases. And if we want to really make it comprehensive, probably for some users, statistics can be stored in some key-value DB - across websites, for the website.